### PR TITLE
Add validations for location and other_locations

### PR DESCRIPTION
**Pull request by GordonDiggs**
Make sure that they match the formats outlined in the spec:
https://github.com/codeclimate/spec/blob/master/SPEC.md#locations

@codeclimate/review

### CODE REVIEW

**Comment by wfleming** [comment]
Nit: we're not validating the `path` of `other_locations` entries, are we? Should we?

**Comment by GordonDiggs** [comment]
@wfleming ah yeah we should. I'll update

**Comment by GordonDiggs** [comment]
@wfleming updated!

File: lib/cc/analyzer/issue_other_locations_format_validation.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      end
+
+      def message
+        "Other locations are not formatted correctly"
+      end
+
+      private
+
+      def other_location_valid?(location)
+        [IssueLocationFormatValidation, IssuePathExistenceValidation, IssuePathPresenceValidation].all? do |klass|
```
Comment (wfleming) [inline lib/cc/analyzer/issue_other_locations_format_validation.rb]: Should we also do the `IssueRelativePathValidation` here?

**Comment by wfleming** [comment]
One note about another validation you probably want in the `other_locations` stack. Otherwise, LGTM.

File: lib/cc/analyzer/issue_other_locations_format_validation.rb (hunk 1 above)
Comment (GordonDiggs) [inline lib/cc/analyzer/issue_other_locations_format_validation.rb]: Yep!

**Pull request merged**